File: results/codes/算法竞赛/doubao/L_3.py
q = int(input())
# Dictionary to store the mapping of rubbish names to their types
rubbish_info = {}

for _ in range(q):
    op = input().split()
    if op[0] == '1':
        # For operation 1, store the rubbish name and its type
        name = op[1]
        type_num = int(op[2])
        if name not in rubbish_info:
            rubbish_info[name] = type_num
    else:
        # For operation 2, find the type based on similarity
        query_name = op[1]
        max_similarity = -1
        min_type = float('inf')
        for stored_name, stored_type in rubbish_info.items():
            similarity = 0
            # Calculate the similarity between the query name and the stored name
            for i in range(min(len(query_name), len(stored_name))):
                if query_name[i] == stored_name[i]:
                    similarity += 1
                else:
                    break
            if similarity > max_similarity:
                max_similarity = similarity
                min_type = stored_type
            elif similarity == max_similarity:
                min_type = min(min_type, stored_type)

        print(min_type)

File: results/codes/算法竞赛/kimi/L_3.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

// Function to calculate the similarity between two strings
int calculateSimilarity(const string& s1, const string& s2) {
    int similarity = 0;
    int minLen = min(s1.size(), s2.size());
    for (int i = 0; i < minLen; ++i) {
        if (s1[i] == s2[i]) {
            similarity++;
        }
    }
    return similarity;
}

int main() {
    int q;
    cin >> q;
    unordered_map<string, int> rubbishType; // Stores the type of each rubbish name
    vector<pair<string, int>> knownRubbish; // Stores all known rubbish names and their types

    while (q--) {
        int op;
        cin >> op;
        if (op == 1) {
            string s;
            int x;
            cin >> s >> x;
            // Update the type in the map
            if (rubbishType.find(s) == rubbishType.end() || x < rubbishType[s]) {
                rubbishType[s] = x;
            }
            // Add the new rubbish name and type to the list
            knownRubbish.push_back({s, x});
        } else if (op == 2) {
            string s;
            cin >> s;
            int maxSimilarity = -1;
            int minType = -1;

            // Check all known rubbish names to find the one with maximum similarity
            for (const auto& [knownName, knownType] : knownRubbish) {
                int similarity = calculateSimilarity(s, knownName);
                if (similarity > maxSimilarity || (similarity == maxSimilarity && knownType < minType)) {
                    maxSimilarity = similarity;
                    minType = knownType;
                }
            }

            // Output the type of the rubbish with maximum similarity
            cout << minType << endl;
        }
    }

    return 0;
}